sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClient.cs

```diff
@@ -231,7 +231,7 @@ public ServiceBusClient(string connectionString, ServiceBusClientOptions options
         {
             Logger.ClientCreateStart(typeof(ServiceBusClient), fullyQualifiedNamespace);
             _options = options?.Clone() ?? new ServiceBusClientOptions();
-            Identifier = DiagnosticUtilities.GenerateIdentifier(fullyQualifiedNamespace);
+            Identifier = _options.Identifier ?? DiagnosticUtilities.GenerateIdentifier(fullyQualifiedNamespace);
```

Comment:
Should the client identifier be threaded through to anything on the connection?

Reply:
Custom identifiers would be passed through down to the connection scope through the options, just made a change so the generated one would be as well.

Reply:
Interesting.. I don't have a strong opinion here and would be inclined to leave as-is for non-custom identifier case. Is this the same for EH?

Reply:
It looks like the `AmqpConnectionScope` identifier in EH is also generated randomly independent of the identifiers of the clients

Reply:
There's no equivalent of `ServiceBusClient` in Event Hubs to own the connection uniquely, and no real benefit from setting it to something known.    For Service Bus, we should probably leave it as-is for now and push the `ServiceBusClient` identifier into the connection if/when we choose to allow durable recovery.

Reply:
@jsquire should this be true for the custom identifier as well? i.e. not setting connection identifiers based on the options.Identifier? Right now I am doing this in `AmqpConnectionScope`

Reply:
It may be worth checking to see what was done in the already GA'ed Service Bus libraries (I think Python). If we don't include the identifier in the connection settings, then the only way it is being used is in logs for creating and disposing the client.

Reply:
In T0, there was a property on the client types for an identifier.   We started adding these back because we've seen a low volume of customer requests along the lines of "I'm migrating and used these properties in my application to know what client this is, since I have several that do different things."

We want the custom identifier on `ServiceBusClient` to be public, but from discussion elsewhere, we don't be using it to set the connection properties just yet.

Reply:
I'd suggest that we add it to the connection if/when we actually need it to avoid confusion.